sqlglot/dialects/teradata.py

```diff
@@ -276,3 +304,25 @@ def createable_sql(self, expression: exp.Create, locations: t.DefaultDict) -> st
                 return f"{this_name}{this_properties}{self.sep()}{this_schema}"
 
             return super().createable_sql(expression, locations)
+
+        def extract_sql(self, expression: exp.Extract) -> str:
+            this = self.sql(expression, "this")
+            if this.upper() != "QUARTER":
+                return super().extract_sql(expression)
+
+            to_char = exp.func("to_char", expression.expression, exp.Literal.string("Q"))
+            return self.sql(exp.cast(to_char, "int"))
+
+        def interval_sql(self, expression: exp.Interval) -> str:
+            multiplier: t.Optional[int] = None
+            unit = expression.text("unit").lower()
```

Comment:
i think there may not be a need to call lower here, since it's always upper case, just change the startswith strings to upper case WEEK etc